include/llbuild/Basic/RedirectedProcess.h

```diff
@@ -0,0 +1,68 @@
+//===- RedirectedProcess.h --------------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2015 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file contains abstractions for creating a process and redirecting
+// input and output of that process.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLBUILD_BASIC_REDIRECTEDPROCESS_H
+#define LLBUILD_BASIC_REDIRECTEDPROCESS_H
+
+#include "llvm/ADT/SmallString.h"
+
+#include <mutex>
+#include <vector>
+
+namespace llbuild {
+namespace basic {
+namespace sys {
+class RedirectedProcess {
```

Comment:
Can we get some documentation comments here on what the class is for and how it behaves?

Some particular questions I would have reading this header w/o the implementation:
1. Is it legal to execute multiple processes on the same object? If not, why have the flags on `execute()` versus the constructor?
2. What are the return values of things like `kill()` presumably false on failure, but always good to call out? I would also call `kill` something more English like `sendSignal`.
3. What does `sigkill()` do?
4. What is the mutex argument for?

This class should probably be non-copyable right? Is there a reason I would want it to be equatable versus working with pointers to it?